src/Servers/Kestrel/Core/src/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -299,5 +299,11 @@ ValueTask<FlushResult> IHttpOutputProducer.WriteChunkAsync(ReadOnlySpan<byte> da
         {
             throw new NotImplementedException();
         }
+
+        public void Complete()
+        {
+            // This can't complete the pipe today because headers NEED to be added as part of the protocol
+            //_dataPipe.Writer.Complete();
```

Comment:
Why is this commented out? Is there a bug somewhere?

Reply:
I also don't get why the _dataPipe.Writer cannot be completed here. Completing it doesn't prevent normal response headers or trailers from being written AFAIK.

Reply:
Wouldn't WriteResponseHeaders need to be called before Complete? Otherwise we won't get normal response headers. 

Maybe we can throw if Complete is called before StartAsync too?